hazelcast/src/test/java/com/hazelcast/client/topic/ClientReliableTopicTest.java

```diff
@@ -56,7 +56,7 @@
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class ClientReliableTopicTest extends HazelcastTestSupport {
+public class ClientReliableTopicTest extends TopicTest {
```

Comment:
I've also noticed we have no tests for BLOCK overload policy. So here's a suggestion, you can do analogous tests for topic, reliable topic, member and client side.

```
@Before
    public void setup() {
        Config config = smallInstanceConfig();
        config.addRingBufferConfig(new RingbufferConfig("blockingReliableTopic*")
                .setCapacity(10)
                .setTimeToLiveSeconds(60));
        hazelcastFactory.newHazelcastInstance(config);

        // for member-side ReliableTopicAbstractTest
        config.addReliableTopicConfig(new ReliableTopicConfig("blockingReliableTopic*")
                .setReadBatchSize(10)
                .setTopicOverloadPolicy(TopicOverloadPolicy.BLOCK));
        
        // for client-side ClientReliableTopicTest
        ClientConfig clientConfig = new ClientConfig()
                .addReliableTopicConfig(new ClientReliableTopicConfig("blockingReliableTopic*")
                        .setReadBatchSize(10)
                        .setTopicOverloadPolicy(TopicOverloadPolicy.BLOCK));
        
        client = hazelcastFactory.newHazelcastClient(clientConfig);
    }
    
        @Test
    public void testBlockingAsync() {
        String randomName = "blockingReliableTopic" + generateRandomString(5);
        AtomicInteger count = new AtomicInteger(0);

        ITopic<Object> topic = client.getReliableTopic(randomName);
        topic.addMessageListener(message -> count.incrementAndGet());


        for (int i = 0; i < 10; i++) {
            topic.publish("message");
        }
        assertTrueEventually(() -> assertEquals(10, count.get()));
        assertCompletesEventually(topic.publishAllAsync(asList("msg 1", "msg 2", "msg 3", "msg 4", "msg 5")).toCompletableFuture());
        assertTrueEventually(() -> assertEquals(15, count.get()));
    }
    
    ```

Reply:
I've created more tests (and extended the existing ones).